File: javascript-drills/js/guessing_game.js
function GuessingGame(answer) {
  this.answer = answer;
  this.solved = false;
}

GuessingGame.prototype.guess = function(playerGuess) {
  this.solved = false;
  if(playerGuess > this.answer){
    return "high";
  } else if(playerGuess < this.answer){
    return "low";
  } else {
    this.solved = true;
    return "correct";
  }
}

GuessingGame.prototype.isSolved = function(){
  return this.solved;
}

File: design-drill-code-style/guessing_game.rb
class GuessingGame
  attr_reader :answer
  attr_accessor :solved

  def initialize(answer)
    @answer = answer
    @solved = false
    @valid_numbers = (1..100).to_a
    raise "Answer must be between 1 and 100" unless @valid_numbers.include? answer
  end

  def guess(number)
    if number == answer
      self.solved = true
      :correct
    elsif (number > answer)
      self.solved = false
      :high
    elsif (number < answer)
      self.solved = false
      :low
    end
  end

  def solved?
    self.solved
  end
end
